qCC/db_tree/ccDBRoot.cpp

```diff
@@ -1687,37 +1690,58 @@ void ccDBRoot::selectByTypeAndName()
 	scDlg.addType("Octree",            CC_TYPES::POINT_OCTREE);
 	scDlg.addType("Kd-tree",           CC_TYPES::POINT_KDTREE);
 	scDlg.addType("Viewport",          CC_TYPES::VIEWPORT_2D_OBJECT);
+	scDlg.addType("Custom Types",      CC_TYPES::CUSTOM_H_OBJECT);
 
 	if (!scDlg.exec())
 		return;
 
-	CC_CLASS_ENUM type = scDlg.getSelectedType();
-	QString name = scDlg.getSelectedName();
+	// for type checking
+	CC_CLASS_ENUM type = CC_TYPES::OBJECT; // all objects are matched by def
+	bool exclusive = false;
+
+	if (scDlg.getTypeIsUsed()) // we are using type checking
+	{
+		type = scDlg.getSelectedType();
+
+		//some types are exclusive, some are generic, and some can be both
+		//(e.g. Meshes)
+		//
+		//For generic-only types the match type gets overridden and forced to
+		//false because exclusive match makes no sense!
+		switch (type)
+		{
+		case CC_TYPES::HIERARCHY_OBJECT: //returned if no type is selected (i.e. all objects are selected!)
+		case CC_TYPES::PRIMITIVE:
+		case CC_TYPES::SENSOR:
+		case CC_TYPES::IMAGE:
+			exclusive = false;
+			break;
+		default:
+			exclusive = scDlg.getStrictMatchState();
+			break;
+		}
+	}
+
 
-	//some types are exclusive, some are generic, and some can be both
-	//(e.g. Meshes)
-	//
-	//For generic-only types the match type gets overridden and forced to
-	//false because exclusive match makes no sense!
-	bool exclusive;
-	switch (type)
+	// for name matching - def values
+	bool regex = false;
+	QString name = "";
```

Comment:
Moreover in the first case `QString::isNull()` will return false.

Of course if you only use `QString::isEmpty()` it's good (it will work in both cases).